homeassistant/components/mobile_app/device_action.py

```diff
@@ -0,0 +1,87 @@
+"""Provides device actions for Mobile App."""
+from typing import List, Optional
+
+import voluptuous as vol
+
+from homeassistant.components import notify
+from homeassistant.components.device_automation import InvalidDeviceAutomationConfig
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_TYPE
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.helpers import config_validation as cv, template
+
+from .const import DOMAIN
+from .util import get_notify_service, supports_push, webhook_id_from_device_id
+
+ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_TYPE): "notify",
+        vol.Required(notify.ATTR_MESSAGE): cv.template,
+        vol.Optional(notify.ATTR_TITLE): cv.template,
+        vol.Optional(notify.ATTR_DATA): cv.template_complex,
+    }
+)
+
+
+async def async_get_actions(hass: HomeAssistant, device_id: str) -> List[dict]:
+    """List device actions for Mobile App devices."""
+    webhook_id = webhook_id_from_device_id(hass, device_id)
+
+    if webhook_id is None or not supports_push(hass, webhook_id):
+        return []
+
+    return [{CONF_DEVICE_ID: device_id, CONF_DOMAIN: DOMAIN, CONF_TYPE: "notify"}]
+
+
+async def async_call_action_from_config(
+    hass: HomeAssistant, config: dict, variables: dict, context: Optional[Context]
+) -> None:
+    """Execute a device action."""
+    webhook_id = webhook_id_from_device_id(hass, config[CONF_DEVICE_ID])
+
+    if webhook_id is None:
+        raise InvalidDeviceAutomationConfig(
+            "Unable to resolve webhook ID from the device ID"
+        )
+
+    service_name = get_notify_service(hass, webhook_id)
+
+    if service_name is None:
+        raise InvalidDeviceAutomationConfig(
+            "Unable to find notify service for webhook ID"
```

Comment:
Should we log the webhook id or is that considered a credential?

Reply:
Yes, it is a credential. They are unique long strings that will be processed by the mobile app integration.